tests/unit/test_charm.py

```diff
@@ -384,3 +394,24 @@ def test_given_self_signed_certificates_not_yet_stored_when_on_certificate_creat
             WaitingStatus("Root Certificates are not yet set"),
             self.harness.charm.unit.status,
         )
+
+    def test_given_self_signed_certificates_with_additional_extensions(self):
+        """Test setting an extension on a CSR and retrieving it from the cert."""
```

Comment:
No need for docstring in all tests (unless they add something not specified in the test name)